test/unit/Cardano/ChainProducer/RustHttpBridge/MockNetworkLayer.hs

```diff
@@ -0,0 +1,74 @@
+module Cardano.ChainProducer.RustHttpBridge.MockNetworkLayer
+    ( mockNetworkLayer
+    ) where
+
+import Prelude
+
+import Control.Monad.Catch
+    ( MonadThrow (..) )
+import Control.Monad.Except
+    ( throwError )
+import qualified Data.ByteString.Char8 as B8
+
+import Cardano.ChainProducer.RustHttpBridge.NetworkLayer
+    ( NetworkLayer (..), NetworkLayerError (..) )
+import Cardano.Wallet.Primitive
+    ( Block (..), BlockHeader (..), Hash (..) )
+import Cardano.Wallet.Slotting
+    ( EpochIndex (..)
+    , LocalSlotIndex (..)
+    , SlotId (..)
+    , slotPrev
+    , slotsPerEpoch
+    )
+
+-- | Embed an epoch index and slot number into a hash.
+mockHash :: SlotId -> Hash a
+mockHash (SlotId (EpochIndex ep) (LocalSlotIndex sl))
+    = Hash $ B8.pack ("Hash " <> show ep <> "." <> show sl)
+
+-- | Extract the epoch index and slot number from a hash.
+unMockHash :: Hash a -> SlotId
+unMockHash (Hash h) = parse . map B8.unpack . B8.split '.' . B8.drop 5 $ h
+    where
+        parse [ep, sl] =
+            SlotId (EpochIndex (read ep)) (LocalSlotIndex (read sl))
+        parse _ = error $ "Could not read mock hash: " ++ B8.unpack h
+
+-- | Create a block header from its hash, assuming that the hash was created
+-- with 'mockHash'.
+mockHeaderFromHash :: Hash a -> BlockHeader
+mockHeaderFromHash h = BlockHeader ep sl prevHash
+    where
+        slotId@(SlotId ep sl) = unMockHash h
+        prevHash = maybe (Hash "?") mockHash (slotPrev slotId)
+
+-- | Generate an entire epoch's worth of mock blocks. There are no transactions
+-- generated.
+mockEpoch :: EpochIndex -> [Block]
+mockEpoch ep = [ Block (mockHeaderFromHash (mockHash sl)) mempty
+               | sl <- [ SlotId ep i | i <- epochs ] ]
```

Comment:
https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#avoid-variable-length-indentation

Probably could do something like:
```hs
mockEpoch ep =
    [ Block (mockHeaderFromHash (mockHash sl)) mempty
    | sl <- [ SlotId ep i | i <- epochs ] ]
```

Reply:
Could you please explain why this indentation is bad?

Reply:
Hi @rvl. I don't think your indentation is bad. It's just that the coding standards specify to not use variable length indentation: https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#avoid-variable-length-indentation

However, if we think this rule is causing us problems, then perhaps we can discuss it in the channel, and make a decision to either keep the rule or scrap it. I don't feel strongly about it either way.